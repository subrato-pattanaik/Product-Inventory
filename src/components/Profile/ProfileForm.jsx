import React, { useState, useEffect, useContext } from "react";
import { withRouter } from "react-router-dom";
import { Col, Form } from "react-bootstrap";
import { useForm } from "react-hook-form";
import { UserContextAPI } from "../ContextAPI/UserContextAPI";
import axios from "axios";

function ProfileForm() {
  const { user } = useContext(UserContextAPI);
  const [editUser, setEditUser] = useState({
    id: 0,
    email: "",
    password: "",
    firstname: "",
    lastname: "",
    location: "",
    mobileno: "",
  });

  const { register, errors, reset } = useForm({
    defaultValues: editUser,
  });

  useEffect(() => {
    axios.get(`http://localhost:4000/users/${user}`).then((res) => {
      setEditUser(res.data);
      reset(res.data);
    });
  }, [user, reset]);

  return (
    <Form autoComplete="off">
      <Form.Row>
        <Form.Group as={Col} className="mr-4">
          <Form.Label>
            Email<span className="text-danger">*</span>
          </Form.Label>
          <Form.Control
            type="email"
            name="email"
            ref={register}
            disabled="true"
          />
          {errors.email && (
            <div className="small lead text-danger mb-2">
              {errors.email.message}
            </div>
          )}
        </Form.Group>

        <Form.Group as={Col}>
          <Form.Label>
            Password<span className="text-danger">*</span>
          </Form.Label>
          <Form.Control
            type="password"
            name="password"
            ref={register}
            disabled="true"
          />
          {errors.password && (
            <div className="small lead text-danger  mb-2">
              {errors.password.message}
            </div>
          )}
        </Form.Group>
      </Form.Row>
      <Form.Row>
        <Form.Group as={Col} className="mr-4">
          <Form.Label>
            First Name<span className="text-danger">*</span>
          </Form.Label>
          <Form.Control name="firstname" ref={register} disabled="true" />
          {errors.firstname && (
            <div className="small lead text-danger  mb-2">
              {errors.firstname.message}
            </div>
          )}
        </Form.Group>

        <Form.Group as={Col}>
          <Form.Label>
            Last Name<span className="text-danger">*</span>
          </Form.Label>
          <Form.Control name="lastname" ref={register} disabled="true" />
          {errors.lastname && (
            <div className="small lead text-danger  mb-2">
              {errors.lastname.message}
            </div>
          )}
        </Form.Group>
      </Form.Row>
      <Form.Row className="mb-3">
        <Form.Group as={Col} className="mr-4">
          <Form.Label>Location</Form.Label>
          <Form.Control name="location" ref={register} disabled="true" />
        </Form.Group>

        <Form.Group as={Col}>
          <Form.Label>
            Mobile No<span className="text-danger">*</span>
          </Form.Label>
          <Form.Control name="mobileno" ref={register} disabled="true" />
          {errors.mobileno && (
            <div className="small lead text-danger  mb-2">
              {errors.mobileno.message}
            </div>
          )}
        </Form.Group>
      </Form.Row>
    </Form>
  );
}

export default withRouter(ProfileForm);
