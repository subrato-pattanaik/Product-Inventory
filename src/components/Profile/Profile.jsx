import React, { useContext } from "react";
import { Container, Row, Col } from "react-bootstrap";
import { withRouter, useHistory } from "react-router-dom";
import ProfileForm from "./ProfileForm";
import { UserContextAPI } from "../ContextAPI/UserContextAPI";

function Profile() {
  const { user } = useContext(UserContextAPI);
  let navigation = useHistory();
  return (
    <>
      {!user ? (
        <>{navigation.push("/Login")}</>
      ) : (
        <Container
          fluid={true}
          className="p-0"
          style={{ minHeight: "89.1vh", background: "#454441" }}
        >
          <Row noGutters>
            <Col>
              <Container className="bg-light my-5">
                <Row className="mx-auto">
                  <Col>
                    <Row>
                      <Col>
                        <h1 className="display-5  px-4 py-2 text-warning text-left font-weight-bold">
                          View your Profile
                        </h1>
                      </Col>
                    </Row>
                    <div className="text-muted border border-warning rounded p-5 mb-5 mx-3">
                      <ProfileForm />
                    </div>
                  </Col>
                </Row>
              </Container>
            </Col>
          </Row>
        </Container>
      )}
    </>
  );
}

export default withRouter(Profile);
