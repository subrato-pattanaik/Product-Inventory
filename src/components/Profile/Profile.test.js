import React from "react";
import { shallow, mount } from "enzyme";
import "mutationobserver-shim";
import Profile from "./Profile";
import { BrowserRouter } from "react-router-dom";
import { Form } from "react-bootstrap";

describe("Profile Page Component", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = mount(
      <BrowserRouter>
        <Profile.WrappedComponent />
      </BrowserRouter>
    );
  });

  it("should test snapshot of Header page", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
