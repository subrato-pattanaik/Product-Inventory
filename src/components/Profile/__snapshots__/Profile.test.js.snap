// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile Page Component should test snapshot of Header page 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "PUSH",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 2,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "7vic5p",
          "pathname": "/Login",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Profile />
  </Router>
</BrowserRouter>
`;
